website/pk-web/dirtrees.go

```diff
@@ -32,22 +31,29 @@ type Directory struct {
 	Dirs     []*Directory // subdirectories
 }
 
-func isGoFile(fi os.FileInfo) bool {
-	name := fi.Name()
-	return !fi.IsDir() &&
+// dirEntry defines functions for determining whether
+// entry is a directory or a file.
+type dirEntry interface {
+	Name() string
```

Comment:
is this a base name only or a complete path?

That is, is this like a fs.FileInfo?

Reply:
LGTM otherwise

Reply:
Only base name i think! if i read it right it resolves to `https://pkg.go.dev/io/fs#DirEntry`

Reply:
Sorry, I mean to document it. For interfaces, it doesn't matter what they actually do. What matters is how they're defined.

Reply:
Ah, i misread it, sorry for the noise!

Reply:
Added comments